social/src/main/java/bisq/social/userprofile/UserProfileService.java

```diff
@@ -126,53 +135,63 @@ public void selectUserProfile(UserProfile value) {
-                     */
 
-    public CompletableFuture<Optional<Entitlement.Proof>> verifyBondedRole(String txId, String signature, PublicKey publicKey) {
-        //todo
-        return CompletableFuture.completedFuture(Optional.of(new Entitlement.BondedRoleProof(txId, signature)));
+    public CompletableFuture<Optional<Entitlement.Proof>> verifyBondedRole(String txId, String signature, PublicKey identityPublicKey) {
+        // inputs: txid, signature from Bisq1
+        // process: get txinfo, grab pubkey from 1st output
+        // verify provided signature matches with pubkey from 1st output and hash of provided identity pubkey
+        return supplyAsync(() -> {
+            try {
+                BaseHttpClient httpClientBsq = getBsqApiHttpClient();
+                BaseHttpClient httpClientBtc = getBtcApiHttpClient();
+                String jsonBsqTx = httpClientBsq.get("tx/" + txId, Optional.of(new Pair<>("User-Agent", httpClientBsq.userAgent)));
+                String jsonBtcTx = httpClientBtc.get("tx/" + txId, Optional.of(new Pair<>("User-Agent", httpClientBtc.userAgent)));
+                checkArgument(BsqTxValidator.initialSanityChecks(txId, jsonBsqTx), "bsq tx sanity checks");
+                checkArgument(BtcTxValidator.initialSanityChecks(txId, jsonBtcTx), "btc tx sanity checks");
+                checkArgument(BsqTxValidator.isBsqTx(httpClientBsq.getBaseUrl(), txId, jsonBsqTx), "isBsqTx");
+                checkArgument(BsqTxValidator.isLockup(jsonBsqTx), "is lockup transaction");
+                String identityKeyHash = Hex.encode(DigestUtil.hash(identityPublicKey.getEncoded()));
+                String signingPubkeyTxt = BtcTxValidator.getFirstInputPubkey(jsonBtcTx);
+                PublicKey signingPubKey = KeyGeneration.generatePublicFromCompressed(Hex.decode(signingPubkeyTxt));
+                boolean signatureMatches = SignatureUtil.verify(Hex.decode(identityKeyHash), Base64.decode(signature), signingPubKey);
```

Comment:
We can extract the DigestUtil.hash(identityPublicKey.getEncoded()) to a var to be reused